CIP-XXXX/README.md

```diff
@@ -0,0 +1,364 @@
+forward.
+
+**Example:**
+
+`[25, 27]`
+
+#### 3. Witness Method
+
+The witness method portion of the payload should describe the method used to generate the witness signature (CIP-8, etc)
+to aid in validation by third parties.
+
+**Example:**
+
+`['CIP-8','Ed25519']`
+
+#### 4. Nonce
+
+As with CIP-36 Voter Registration method, we must include a piece of data that changes to avoid a replay attack. As with
+CIP-36, and stake pool KES counter rotation, the nonce is expected to be an integer value that is incremented as new or
+updated registrations occur. The suggestion to use the current slot tip of the network at the time of submission from 
+CIP-36 is also recommended here.
+
+Validators and indexers observing token registrations should always honor the verified registration with the highest
+nonce value.
+
+**Example:**
+
+`12345`
+
+### Optional Fields
+
+The following are optional fields that may be included in the token registration to provide additional information or
+context to the registration.
+
+#### 5. Data Oracle URI
+
+To be utilized and expanded upon in a separate CIP, this should be a valid URI pointing to a source of additional,
+potentially dynamic information relating to the project and/or the tokens minted under it.
+
+**Example:**
+
+`'https://oracle.mytokenproject.io/'`
+
+#### 6. CIP-Specific Information
+
+This entry, if present, should be a CIP ID indexed object containing additional information pertaining to that CIP.
+These additional CIPs/Schemas to be determined by the community could include:
+
+- CIP-25/68 NFT Project top-level project information
+- CIP-26 FT Project monetary policy information
+
+**Example:**
+
+```json 
+{
+  "25": {
+    "project_name": "Cool NFT Project",
+    "description": [
+      "This is a description of my project", 
+      "longer than 64 characters so broken up into a string array"
+    ],
+    "banner_image_uri": "https://static.coolnftproject.io/images/banner1.jpg",
+    "icon_image_uri": "https://static.coolnftproject.io/images/icon.png",
+    "social_media": {
+      "discord": "https://coolnftproject.discord.gg/",
+      "twitter": "https://twitter.com/coolNFTproject"
+    }
+  },
+  "27": {
+    "rate": "0.05",
+    "addr": [
+      "addr_test1qqp7uedmne8vjzue66hknx87jspg56qhkm4gp6ahyw7kaahevmtcux",
+      "lpy25nqhaljc70094vfu8q4knqyv6668cvwhsq64gt89"
+    ]
+  }
+}
+```
+
+### Token Registration Witness Array
+
+The Witness Array included in the on-chain metadata should consist of an array of arrays with two elements, the public
+key hash of the signing key (as included in the native script) and the signed key witness. If a script requires multiple
+signatures, enough signatures to meet the criteria of the script should be included and required for proper validation
+of an updated token registration.
+
+**Example**
+
+```json 
+[
+  [
+    "e97316c52c85eab276fd40feacf78bc5eff74e225e744567140070c3j",
```

Comment:
Can you submit binary via on-chain metadata? Wouldn't it need to be string-encoded to match the JSON format?

Reply:
I would at least write the spec to allow for either. json is a limitation of building a transaction with cardano-cli and not the on-chain metadata itself. You can save 50% of the size by not using hex strings. CIP-25 v2 has been updated to allow binary strings.

Reply:
I will take a look and revise accordingly!

Reply:
> Can you submit binary via on-chain metadata? Wouldn't it need to be string-encoded to match the JSON format?

Yes of course! And at this point I don't know why people keep submitting hex-encoded strings to the chain instead of plain byte strings. Hex-encoded data are twice the size and less efficient to decode...

Reply:
CDDL v1.0.0 spec has been published noting that signature payloads should be arrays of byte strings rather than strings.

Reply:
Marking this as resolved as examples and CDDL have been updated to use binary strings rather than hex strings.